.github/workflows/nightly.yml

```diff
@@ -0,0 +1,53 @@
+---
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+      - with-extensions
+  schedule:
+    - cron: "30 0 * * *"
+
+jobs:
+  test-setup:
+    strategy:
+      fail-fast: false
+      matrix:
+        # For info on the OS versions, architecture, etc., see
+        # https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories
+        os: [ubuntu-latest, ubuntu-24.04-arm, macos-13, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout with-extensions branch
+        uses: actions/checkout@v4
+        with:
+          ref: with-extensions
+
+      - name: Get latest commit hash
+        id: with-extensions-rev
+        run: |
+          HASH=$(git rev-parse HEAD)
+          echo "hash=${HASH}" >> "$GITHUB_OUTPUT"
+
+      - name: Install autoconf on macOS
+        if: startsWith(matrix.os, 'macos')
+        run: brew install autoconf
+
+      - uses: ocaml/setup-ocaml@v3
+        with:
+          # Invalidate cache when with-extensions has new commits, since the
+          # archive URL in the opam file for 5.2.0+flambda2 might have changed.
+          cache-prefix: ${{ steps.with-extensions-rev.outputs.hash }}
+          ocaml-compiler: ocaml-variants.5.2.0+flambda2
+          opam-pin: false
+          opam-repositories: |
+            with-extensions: git+https://github.com/janestreet/opam-repository.git#with-extensions
```

Comment:
This isn't actually using the with-extensions checkout done in the step above.

Reply:
Yes, but you need to also use that for the opam repo above, or else you won't get the right revision (for example if it's a PR, this would just build the `with-extensions` branch and not the PR contents.

Reply:
Yes, true. The PR was intended to be run as a nightly script, right now, but could lead to future bugs. I'll fix it.